src/Chapter3.hs

```diff
@@ -371,8 +377,31 @@ after the fight. The battle has the following possible outcomes:
  ⊛ Monster defeats the knight. In that case return -1
  ⊛ Neither the knight nor the monster wins. On such an occasion, the knight
    doesn't earn any money and keeps what they had before.
-
 -}
+data Knight = MkKnight
+  { knightHealth        :: Int,
+    knightAttackDamage  :: Int,
+    knightGoldHeld      :: Int
+  } deriving (Show)
+
+data Monster = MkMonster
+  { monsterHealth        :: Int,
+    monsterAttackDamage  :: Int,
+    monsterGoldHeld      :: Int
+  } deriving (Show)  
+
+fight :: Monster -> Knight -> Int
+fight monster knight
+  | kDmg >= mHealth = kGold + mGold
+  | mDmg >= kHealth = -1
+  | otherwise = kGold
+  where
+    kDmg = knightAttackDamage knight
+    mDmg = monsterAttackDamage monster
+    kHealth = knightHealth knight
+    mHealth = monsterHealth monster
+    kGold = knightGoldHeld knight
+    mGold = monsterGoldHeld monster
```

Comment:
Wow, love this feature. I think `{-# OPTIONS_GHC -Werror=missing-fields #-}` is a much better solution than adding `!` to every record field type - don't you think? (IIUC one can go one way or the other?)

Reply:
@saatvikshah it is considered a best-practice to use strict fields in data types anyway, so not a problem generally 🙂 You can even go one step further and enable `{-# LANGUAGE StrictData #-}` in modules to avoid using `!` before each type. There are very few cases where you really want to have lazy fields, so such cases should be thought through and can be marked explicitly.

Anyway, on CI you usually want to enable `-Werror` for all warnings to avoid having all warnings in your code at all, so no need to specify `-Werror` for this particular warning